webpack/assets/javascripts/react_app/redux/actions/bookmarks/bookmarks.test.js

```diff
@@ -9,36 +8,42 @@ import {
   onFailureActions,
   onSuccessActions,
 } from './bookmarks.fixtures';
-import { bookmarks } from '../../reducers/bookmarks/bookmarks.fixtures';
 import API from '../../../API';
+import { mockRequest, mockReset } from '../../../mockRequests';
 
 const middlewares = [thunk];
 const mockStore = configureMockStore(middlewares);
+const store = mockStore(initialState);
+
+afterEach(() => {
+  store.clearActions();
+  mockReset();
+});
 
 describe('bookmark actions', () => {
   it('should handle failure to load bookmarks', () => {
-    const store = mockStore(initialState);
-    const { url, controller } = requestData;
+    const { url, controller, searchRegex } = requestData;
 
-    store.dispatch(actions.getBookmarks(url, controller));
-    expect(store.getActions()).toEqual(onFailureActions);
+    mockRequest({
+      searchRegex,
+      status: 422,
+    });
+    return store.dispatch(actions.getBookmarks(url, controller))
+      .then(() => expect(store.getActions()).toEqual(onFailureActions));
```

Comment:
maybe there is a better way to avoid promises? I like the approach of overall but I see this as a potential limitation to writing more fully scoped tests...will follow up in an email shortly to you guys.

Reply:
@priley86 +1
Checkout this gist:
https://gist.github.com/sharvit/cf658a8494758be8062a63121b925fdc

This is kind of a solution for unit-testing i tried to implement.
We have been discussing about it quit a lot at the last month.
We feel we should migrate the tests soon to this style and start writing integration tests.

Reply:
This approach is suggested by redux ( Async Action Creators part):
https://redux.js.org/docs/recipes/WritingTests.html
I agree that we should have some test conventions, especially when it involves integration between react component and redux, I've started to work on that kind of integration helper.

Reply:
Again we have same dilemma....

```
export const submitForm = ({
  item, url, values, method = 'post',
}) => {
  verifyProps(item, values);
  return dispatch =>
    API[method](url, values)
      .then(({ data }) => {
        dispatch({
          type: `${item.toUpperCase()}_FORM_SUBMITTED`,
          payload: { item, data },
        });
```
... I am chaining promises.... do you not prefer it to be `gen.next()` ?

Reply:
I <3 the use of Jest snapshots though.. maybe we can incorporate it with Saga too... just a thought.

Reply:
+1 on jest snapshots,
I think we should have a deeper look on saga, particularly if its benefit worth the migration from thunk.  I don't sure if we have complicated API requests (which require chaining promises or so) in foreman, testing would be much easier though.

Reply:
leaving saga aside, any thoughts on redux-promise middlewere?
It could answers your concern with promises and yet simpler to migrate.

Reply:
where does this leave this PR? I'm not sure i fully understand its status?

Reply:
I don't think it relevant to this PR, @priley86  has mentioned good points that we should look into.